pandas/core/generic.py

```diff
@@ -1303,8 +1303,59 @@ def __invert__(self):
 
     def equals(self, other):
         """
-        Determines if two NDFrame objects contain the same elements. NaNs in
-        the same location are considered equal.
+        Test whether two DataFrame objects contain the same elements.
+
+        This function allows two DataFrame objects to be compared against
+        each other to see if they same shape and elements. NaNs in the same
+        location are considered equal. The column headers do not need to have
+        the same type, but the elements within the columns must be
+        the same dtype.
+
+        Parameters
+        ----------
+        other : DataFrame
+            The other DataFrame to be compared with the first.
+
+        Returns
+        -------
+        bool
+            True if all elements are the same in both DataFrames, False
+            otherwise.
+
+        See Also
+        --------
+        pandas.Series.eq : Compare two Series objects of the same length
```

Comment:
No need for the `pandas.` prefix, just `Series.eq` is enough.

I think we need `eq` and `equals` of both `Series` and `DataFrame`, not just `Series.eq`.

Reply:
Would it be redundant to mention `equals` since this is the docstring for that method?